File: Seminar/src/main/java/org/sopt/Seminar/domain/category/service/CategoryService.java
package org.sopt.Seminar.domain.category.service;

import lombok.RequiredArgsConstructor;
import org.sopt.Seminar.domain.category.exception.CategoryNotFoundException;
import org.sopt.Seminar.domain.category.model.Category;
import org.sopt.Seminar.domain.category.model.CategoryId;
import org.sopt.Seminar.domain.category.repository.CategoryRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class CategoryService {

    private final CategoryRepository categoryRepository;

    public Category getByCategoryId(CategoryId categoryId) {
        return categoryRepository.findById(categoryId).orElseThrow(
                CategoryNotFoundException::new);
    }
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/domain/category/service/CategoryService.kt
package org.sopt.kotlinsemina.domain.category.service

import org.sopt.kotlinsemina.domain.category.exception.CategoryNotFoundException
import org.sopt.kotlinsemina.domain.category.model.Category
import org.sopt.kotlinsemina.domain.category.model.CategoryId
import org.sopt.kotlinsemina.domain.category.repository.CategoryRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.lang.IllegalArgumentException

@Service
@Transactional(readOnly = true)
class CategoryService(
    private val categoryRepository: CategoryRepository
) {
    fun getByCategoryId(categoryId: CategoryId): Category {
        return categoryRepository.findByIdOrNull(categoryId) ?: throw CategoryNotFoundException()
    }
}